Sources/Layout/MessageSizeCalculator.swift

```diff
@@ -49,6 +49,23 @@ open class MessageSizeCalculator: CellSizeCalculator {
 
     public var incomingMessageBottomLabelAlignment = LabelAlignment(textAlignment: .left, textInsets: UIEdgeInsets(left: 42))
     public var outgoingMessageBottomLabelAlignment = LabelAlignment(textAlignment: .right, textInsets: UIEdgeInsets(right: 42))
+    
+    private lazy var textContainer: NSTextContainer = {
+        let textContainer = NSTextContainer()
+        textContainer.maximumNumberOfLines = 0
+        textContainer.lineFragmentPadding = 0
```

Comment:
@zhongwuzw This is where I left off on reviewing this PR. I was a little concerned that the values used in the `textContainer` wouldn't match up with the `textContainer` used in `MessageLabel`. Let's say that the user changes the `MessageLabel` to have a `maximumNumberOfLines` to 2 or `lineFragmentPadding` to 3. What do you think?

Reply:
@SD10 Yeah, I also consider this issue, I prefer to let user override size calculation method.But also we can, if user change property of `Label`, provide `maximunNumberOfLines` `lineFragmentPadding` to the layout manager? After that we can calculate specific size for each label? 🤔

Reply:
I don't think there's a good way for the `MessageLabel` and the `MessageSizeCalculator` to communicate within the framework. I'm really worried that this approach will lead to a large amount of undefined behavior. 

We could possibly make a delegate that provides a shared `MessageSizeConfiguration` object to both the`MessagesCollectionViewFlowLayout` _and_ the `MessageLabel` but blahhh 🤢

Reply:
@SD10 Add a shared `configuration` between `theMessagesCollectionViewFlowLayout` and `MessageLabel` seems weird 😂 , I still think users may should override `messageContainerSize(for message: MessageType)->CGSize` to return size, and don't change properties of `MessageLabel` directly, instead to reload cell. 🤔